go/vt/vttablet/tabletmanager/vreplication/utils.go

```diff
@@ -232,6 +232,16 @@ func isUnrecoverableError(err error) bool {
 		sqlerror.ERWrongValueCountOnRow:
 		log.Errorf("Got unrecoverable error: %v", sqlErr)
 		return true
+	case sqlerror.ERErrorDuringCommit:
+		switch sqlErr.HaErrorCode() {
+		case
+			sqlerror.HaErrCrashed,
+			sqlerror.HaErrTooBigRow,
+			sqlerror.HaErrFoundDuppUnique,
+			sqlerror.HaErrDiskFullNowait:
+			log.Errorf("Got unrecoverable error: %v", sqlErr)
+			return true
+		}
```

Comment:
@shlomi-noach Isn't it easier to invert this and only here have explicit error codes that are recoverable? That's I think also more robust for new error codes etc. in the future to assume it's not recoverable?

Reply:
I'm ambivalent. On one hand, until today I have never encountered this error in Vitess. On the other hand, I'm afraid to introduce new (so called "unaccounted for") failure points.

Reply:
I'd lean towards treating unknown things then as not recoverable / retryable? Seems less risky to keep hammering on something that is an unexpected / unknown failure?

Reply:
We now opt-in for recoverable errors.